### DISCUSSION TITLE
[Feature] Provide platform support for JetBrains Space

### DISCUSSION THREAD

**Discussion by dragetd** [renovatebot/renovate#25021] [Suggest an Idea]
**What would you like Renovate to be able to do?**

JetBrains is building a new development platform targeted at a professional audience called 'Space'. See
https://www.jetbrains.com/space/

It provides a documented and powerful API that should cover all needs. To see the API documentation, one needs an account.

It would be nice if renovate would support this API 'platform: space' and allow it to be run against a Space.

Thank you for the awesome project!

**Comment by zharinov** [comment]
The API documentation is available via `HTTP API Playground` (sign-up required)

**Comment by misterkramer** [comment]
JFYI. 
API Reference is also available here - https://www.jetbrains.com/help/space/http-api-reference.html

**Comment by vooft** [comment]
Hello, I've been playing with the API and built an experimental integration of renovate with Space: https://github.com/vooft/renovate/tree/space

I'm currently adding more tests, but overall it can check out a repo, create PRs, update description, adding reviewers/assignees, automerge, etc.
Working with issues is not supported yet and some features are simply missing in Space (like labels, for example).

To make it work you need to create a personal token in Preferences and add it to your config.
Minimal config will look like:
```js
module.exports = {
  token: '<your-personal-token>',
  repositories: ['main/my-repo'], // goes like <project-key>/<repo-name>, default project key is "MAIN", but it seems that it is case-insensitive
  platform: 'space',
  endpoint: '<your-org>.jetbrains.space',
};
```

There are still some improvements to be made, like right now urls are being created using string templates, not taking in account url encoding.

**Reply by leabaertschi** [reply]
Is there any news on this?

**Reply by vooft** [reply]
JetBrains discontinued Space completely, including the planned replacement https://blog.jetbrains.com/space/2024/11/27/discontinuation-of-the-spacecode-private-preview/

**Reply by leabaertschi** [reply]
I know, unfortunately, we're stuck with it for the moment :(

**Reply by viceice** [reply]
we won't accept a new platform which is already deprecated. sorry but the maintenance burden is too much.

**Reply by leabaertschi** [reply]
Totally understandable. I was just wondering if there was something that we can use as long as we're stuck with space, seeing that this post is half a year old.
